abba/json/include/Parse.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef PARSE_HPP
+#define PARSE_HPP
+
+#include "Json.hpp"
+#include <stdlib.h>
+
+class Parse
+{
+private:
+	std::fstream	*_file;
+	size_t			_prevPos;
+	Json			*_root;
+	Json			*_current;
+	Json			json;
+
+private:
+	enum jsonType
+	{
+		CURLY_OPEN,
+		CURLY_CLOSE,
+		STRING,
+		NUMBER,
+		BOOLEAN,
+		COLON,
+		ARRAY_OPEN,
+		ARRAY_CLOSE,
+		COMMA,
+		NULL_TYPE
+	};
+
+private:
+	struct Type
+	{
+		std::string	value;
+		jsonType	type;
+		jsonType	nextToken;
+	};
+
+public:
+	char	getWithoutWhiteSpace();
+	Type	getToken();
+	bool	hasMoreToken();
+
+public:
+	Parse(std::string filename);
```

Comment:
It would make more sense to take something like a `std::istream&` here, that way you leave the file handling (opening, closing, etc) to another part of the program and you can also handle any other stream such as `std::cin` (or my TcpStream for sockets 😉 )